tools/plugins/com.liferay.ide.portlet.core/src/com/liferay/ide/portlet/core/spring/dd/SpringPluginPackageDescriptorHelper.java

```diff
@@ -0,0 +1,153 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.portlet.core.spring.dd;
+
+import com.liferay.ide.core.ILiferayConstants;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.StringPool;
+import com.liferay.ide.portlet.core.IPluginPackageModel;
+import com.liferay.ide.portlet.core.PluginPropertiesConfiguration;
+import com.liferay.ide.portlet.core.PortletCore;
+import com.liferay.ide.portlet.core.dd.PluginPackagesDescriptorHelper;
+
+import java.io.File;
+import java.io.FileWriter;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.wst.common.frameworks.datamodel.IDataModel;
+
+/**
+ * @author Terry Jia
+ */
+public class SpringPluginPackageDescriptorHelper extends PluginPackagesDescriptorHelper
+{
+
+    public SpringPluginPackageDescriptorHelper()
+    {
+        super();
+    }
+
+    public SpringPluginPackageDescriptorHelper( IProject project )
+    {
+        super( project );
+    }
+
+    @Override
+    public IStatus doAddNewPortlet( IDataModel dataModel )
+    {
+        if( canAddNewPortlet( dataModel ) )
+        {
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "aopalliance.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-aop.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-beans.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-context.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-core.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-expression.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-web.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-webmvc.jar" );
+            addPortalDependency( IPluginPackageModel.PROPERTY_PORTAL_DEPENDENCY_JARS, "spring-webmvc-portlet.jar" );
+        }
+
+        return Status.OK_STATUS;
+    }
+
+    public IStatus addPortalDependency( String propertyName, String value )
+    {
+        if( CoreUtil.isNullOrEmpty( value ) )
```

Comment:
This code looks like it is copied from Vaadin wizard, but it does the same thing, so we should not repeat that code here again but should somehow reuse it.